cl/lib/elasticsearch_utils.py

```diff
@@ -3241,15 +3237,41 @@ def do_es_sweep_alert_query(
     if child_query:
         rd_results = responses[2]
 
+    # Re-run parent query to fetch potentially missed docket IDs due to large
+    # result sets.
+    should_repeat_parent_query = (
+        docket_results and docket_results.hits.total.value >= 10_000
+    )
+    if should_repeat_parent_query:
+        docket_ids = [int(d.docket_id) for d in main_results]
+        # Adds extra filter to refine results.
+        parent_query.filter.append(Q("terms", docket_id=docket_ids))
+        parent_search = search_query.query(parent_query)
+        parent_search = parent_search.source(includes=["docket_id"])
+        docket_results = parent_search.execute()
+
     limit_inner_hits({}, main_results, cd["type"])
     set_results_highlights(main_results, cd["type"])
 
-    for result in main_results:
-        child_result_objects = []
-        if hasattr(result, "child_docs"):
-            for child_doc in result.child_docs:
-                child_result_objects.append(child_doc.to_dict())
-            result["child_docs"] = child_result_objects
+    # This block addresses a potential issue where the initial child query
+    # might not return all expected results, especially when the result set is
+    # large. To ensure complete data retrieval, it extracts child document IDs
+    # from the main results and refines the child query filter with these IDs.
+    # Finally, it re-executes the child search.
+    should_repeat_child_query = (
+        rd_results and rd_results.hits.total.value >= 10_000
+    )
+    if should_repeat_child_query:
+        rd_ids = [
+            int(rd.to_dict()["id"])
```

Comment:
I did some tests, and the correct way to get the ID here to avoid a an error should be:

```suggestion
            int(rd["_source"]["id"])
```
no need to transform it into a dict first

Reply:
alright!